### ISSUE TITLE
libgccjit not found after gcc update

### ISSUE THREAD

**Issue by R0flcopt3r** [d12frosted/homebrew-emacs-plus#483]
Please make sure to follow these steps (and mark the checkboxes):

- [x ] run `brew update` and try to reproduce the issue again
- [ x] run `brew doctor`, fix all issues and try to reproduce your issue again
- [ x] run `brew config` and `brew doctor` and include their output

## What you were trying to do

Trying to get emacs working after a update to 12.5 and latest commandline tools.

## What happened (include command output)

fails to locate `libgccjit` library. It is installed via brew as well.

<pre>
$ brew install emacs-plus@29 --with-no-frame-refocus --with-native-comp --with-cacodemon-icon
</pre>

<details>
  <summary>Command output</summary>
  <pre>

==> Cloning https://github.com/emacs-mirror/emacs.git
Updating /Users/eirik/Library/Caches/Homebrew/emacs-plus@29--git
==> Checking out branch master
Already on 'master'
Your branch is up to date with 'origin/master'.
HEAD is now at 9b630dc252 Remove more XEmacs compat code from viper-util.el
==> Installing emacs-plus@29 from d12frosted/emacs-plus
==> Patching
==> Applying no-frame-refocus-cocoa.patch
patching file src/frame.c
Hunk #1 succeeded at 2094 with fuzz 2 (offset 38 lines).
==> Applying fix-window-role.patch
patching file src/nsterm.m
Hunk #1 succeeded at 9385 (offset 617 lines).
==> Applying system-appearance.patch
patching file src/frame.h
patching file src/nsfns.m
Hunk #1 succeeded at 1377 (offset 121 lines).
patching file src/nsterm.m
Hunk #1 succeeded at 1933 (offset 44 lines).
Hunk #2 succeeded at 5777 (offset 382 lines).
Hunk #3 succeeded at 6065 (offset 424 lines).
Hunk #4 succeeded at 6115 (offset 424 lines).
Hunk #5 succeeded at 6341 (offset 424 lines).
Hunk #6 succeeded at 9718 (offset 816 lines).
Hunk #7 succeeded at 11052 (offset 994 lines).
==> ./autogen.sh
==> ./configure --enable-locallisppath=/opt/homebrew/share/emacs/site-lisp --infodir=/opt/homebrew/Cellar/emacs-plus@29/29.0.50/share/info/emacs --prefix=/opt/homebrew/Cellar/emacs-plus@29/29.0.50 --with-xml2 --with-gnutls --with-native-compilation --without-dbus --without-imagemagick --with-modules --with-rsvg --with-
Last 15 lines from /Users/eirik/Library/Logs/Homebrew/emacs-plus@29/02.configure:
checking for libsystemd >= 222... no
checking for jansson >= 2.7... yes
checking for sys/inotify.h... no
checking for libkqueue... no
checking for library containing kqueue... none required
checking for jpeglib 6b or later... -ljpeg
checking for lcms2... yes
checking for library containing inflateEnd... -lz
checking for dladdr... yes
checking for dlfunc... no
checking for gcc_jit_context_acquire in -lgccjit... no
configure: error: ELisp native compiler was requested, but libgccjit was not found.
Please try installing libgccjit or a similar package.
If you are sure you want Emacs be compiled without ELisp native compiler,
pass the --without-native-compilation option to configure.

Do not report this issue to Homebrew/brew or Homebrew/core!

These open issues may also help:
Could not install emacs-plus@28 because tar can't find xz  https://github.com/d12frosted/homebrew-emacs-plus/issues/481
Could not install with ``brew install emacs-plus@29 --with-native-comp --with-elrumo2-icon`` https://github.com/d12frosted/homebrew-emacs-plus/issues/473

  </pre>
</details>

## Output of `brew config`

<pre>
$ brew config
HOMEBREW_VERSION: 3.5.7
ORIGIN: https://github.com/Homebrew/brew
HEAD: ce8ef89ec0cfe875c48c3a83a843af9074c05930
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 2d6a477140adb85f24689d844942180824ae5424
Core tap last commit: 50 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.dDxxMBpdwl/org.xquartz:0
HOMEBREW_EDITOR: emacsclient -c --alternate-editor=emacs
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 13.1.6 build 1316
Git: 2.37.1 => /opt/homebrew/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.5-arm64
CLT: 13.4.0.0.1.1651278267
Xcode: N/A
Rosetta 2: false
</pre>

## Output of `brew doctor`

<pre>
$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Putting non-prefixed coreutils in your path can cause GMP builds to fail.

Warning: Putting non-prefixed findutils in your path can cause python builds to fail.
</pre>

**Comment by R0flcopt3r** [comment]
When ommiting `--with-native-comp` i get this error *most* times. 
```==> gmake
Last 15 lines from /Users/eirik/Library/Logs/Homebrew/emacs-plus@28/03.gmake:
  "_rpl_optind", referenced from:
      _main in etags-0159c2.o
  "_rpl_re_compile_pattern", referenced from:
      _add_regex in etags-0159c2.o
  "_rpl_re_match", referenced from:
      _readline in etags-0159c2.o
  "_rpl_re_search", referenced from:
      _find_entries in etags-0159c2.o
  "_rpl_re_set_syntax", referenced from:
      _add_regex in etags-0159c2.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
gmake[1]: *** [Makefile:376: etags] Error 1
gmake[1]: Leaving directory '/private/tmp/emacs-plusA28-20220803-19449-k1scxs/emacs-28.1/lib-src'
gmake: *** [Makefile:436: lib-src] Error 2```

I had one successful install without native comp, but decided to reinstall right after to see if i had somehow fixed my issues, but now i am unable to install any version of emacs-plus.

**Comment by noeldes** [comment]
I got this yesterday when I brew uninstalled and reinstalled d12's v29, which had built in the past.

In my case, for whatever reason, removing all flags aside from  --with-native-comp worked, in the same terminal session that had just failed.

Give that a try?

**Comment by d12frosted** [comment]
Hey folks,

I am not available till the next week, so can't really look into this issue.

As a workaround, try uninstalling `gcclibjit` and giving it another spin.

**Comment by cdlm** [comment]
Uninstalling `libgccjit` just means brew will install it again as a dep of `emacs-plus`. I don't see the point.
What I see on my machine is that `emacs-plus@29` sometimes builds sometimes not, often at minutes intervals (build fails, relaunch it immediately and it passes). To me that suggests some spooky action at a distance like a side effect via the environment or some temp file lingering around but I have no clue how do debug that.

The `emacs-plus@28` formula has been a tiny bit more stable but it's also flaky.

**Comment by d12frosted** [comment]
> Uninstalling libgccjit just means brew will install it again as a dep of emacs-plus. I don't see the point.

That's the intention - to reinstall `libgccjit`. Homebrew has a notion of versions and revisions. When revision is bumbed, all dependant formulas are **not** rebuilt and sometimes that breaks dependant software, so you have to manually reinstall stuff. I remember it was a constant case with `imagemagick` and I've seen some cases with `libgccjit` where reinstalling helped. Though in the case with `libgccjit` I don't remember the reason, maybe it was something different.

> What I see on my machine is that emacs-plus@29 sometimes builds sometimes not, often at minutes intervals (build fails, relaunch it immediately and it passes). To me that suggests some spooky action at a distance like a side effect via the environment or some temp file lingering around but I have no clue how do debug that.

I would suspect that it's related to env changes I did some time ago. But then you say that Emacs+ 28 is also flaky, so it's not the case. In general, `nativecomp` is highly unstable thing. It constantly breaks on one machine, while working on others. Time to time it breaks because of updates of gcc/libgccjit. 🤷 

I have returned from small vacation and will take a deeper look into this case as I see lots of messages related to nativecomp breakage. 

P.S. I see some duplicating issues, so I will eventually close all of them except for one and will keep that untill the issue is fixed.

**Comment by d12frosted** [comment]
I know this issue came before #485, but the latter has more information with some workarounds, so I am going to keep that. Please use #485 to report anything related to libgccjit discovery.

Fix for Emacs 28 is in master, fix for Emacs 29 is waiting for CI - #492.

**Comment by d12frosted** [comment]
#492 is merged. In order to test it, performa a combination of `brew update` and `brew install emacs-plus@29 --with-native-comp [options]`. Let me know in #485 if it doesn't work.
